libs/api/libbpf_map.cpp

```diff
@@ -246,36 +252,55 @@ bpf_object__find_map_fd_by_name(const struct bpf_object* obj, const char* name)
 int
 bpf_map__set_pin_path(struct bpf_map* map, const char* path)
 {
+    if (!map) {
+        return libbpf_result_err(EBPF_INVALID_ARGUMENT);
+    }
```

Comment:
The real libbpf always AV's here instead of returning an error that the caller might ignore.  As discussed in triage meeting, it's better to crash, and at least in this case, libbpf does so, so I don't see a reason to be different.